### PR TITLE
ISPN-15659 Hot Rod Client RemoteCache#computeAsync() can lose the result of remapping processing when executed concurrently against a new entry

### PR DESCRIPTION
**Pull request by jabolina**
https://issues.redhat.com/browse/ISPN-15659

### CODE REVIEW

**Comment by tristantarrant** [comment]
It's :green_heart:

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/HotRodIntegrationTest.java [hunk 1]
```diff
@@ -351,6 +355,52 @@ public void testPutSerializableByteArray() {
...
+            checkPoint.trigger("computed_first");
+            try {
+               checkPoint.awaitStrict("computed_first_proceed", 10, TimeUnit.SECONDS);
+            } catch (InterruptedException | TimeoutException e) {
+               throw new RuntimeException("Failed receiving event", e);
+            }
+         }
+         return value;
+      });
+
```
Comment (wburns) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/HotRodIntegrationTest.java:380]: We need to add a `checkpoint.awaitStrict("computed_first", 10, TimeUnit.SECONDS` here to make sure the other remote call is inside the compute block.

**Review by wburns — CHANGES_REQUESTED** [review]
LGTM, just a simple test update to make sure it won't become flaky.

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/HotRodIntegrationTest.java (hunk 1 above)
Comment (jabolina) [reply client/hotrod-client/src/test/java/org/infinispan/client/hotrod/HotRodIntegrationTest.java:380]: Good catch!

**Review by wburns — APPROVED** [review]
LGTM, just waiting on CI.. can it be green again? :)

**Comment by ryanemerson** [comment]
Thanks @jabolina

**Pull request merged**